comparisons/comparison-bpmn.md

```diff
@@ -312,10 +312,11 @@ version: '1.0'
 start: SubflowRepeat
 states:
 - name: SubflowRepeat
-  type: subflow
-  workflowId: dosomethingandwaitformessage
-  repeat:
-    max: 10
+  type: foreach
+  inputCollection: ${ [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ] }
```

Comment:
ForEach state is incorrect here imo. 
Foreach state executes each iteration in parallel - https://github.com/serverlessworkflow/specification/blob/main/specification.md#foreach-state --> "Each iteration of the ForEach state should be executed in parallel."

where "repeat" is a loop one at a time

also in jq you can use "[range(0;9)]" as the expression instead of creating the array manually 

So I think foreach state is not a replacement for repeat

Reply:
also i think this implies that input to the inputCollection is an empty object where as the input to the inputCollection is the state data 
(meaning the expression in the pr or using "range" only work on "{ }", they don't just generate an array out of the blue)

Reply:
just noticed the bpmn image is incorrect, in that it shows the multi-instance subprocess instead of the loop marker. That probably caused the confusion, sorry. Will update the image asap

Reply:
You ignored that I put `max: 1`.  This makes it serial in nature...

Reply:
ok i see, missed it. i guess then the issue left is  that data input for each subflow exec here is the element of the input collection (1, 2, 3, 4 ...) and not state data as intended by business logic in the initial example.

Reply:
i think i removed a comment here too somehow :( 
inputCollection expression data input is the state data, so i think it cannot be an expression that somehow creates the data input for the subflow executions. either way the expressions have to operate on some data input and i assume the intend was that they somehow operate on "{ }" maybe, idk. for example it guess just like ${ .orders } or even ${ .inputArray } should be fine for users to undertand

Reply:
@tsurdilo I'm not sure I follow what you're saying in your last comment.

I can adjust the expression so that it's iterating 10 times and the input to each iteration is the state input with the following, I believe:

```yaml
inputCollection: ${ reduce range(0,10) as $d ([.]; .[$d+1] += .[0] ) }
```

That's probably not very easy to understand at first glance though.   Are you suggesting that I just change the example to say there's an array of 'orders' and the example just iterates over those 'orders'?

Reply:
Got more clarity, will replace 'repeat' usage in examples with explicit control-flow logic.